src/modules/UI/scenes/SendConfirmation/action.js

```diff
@@ -34,45 +37,54 @@ export const UPDATE_CRYPTO_AMOUNT_REQUEST = PREFIX + 'UPDATE_CRYPTO_AMOUNT_REQUE
 export const USE_MAX_CRYPTO_AMOUNT = PREFIX + 'USE_MAX_CRYPTO_AMOUNT'
 export const UPDATE_PARSED_URI = PREFIX + 'UPDATE_PARSED_URI'
 export const UPDATE_TRANSACTION = PREFIX + 'UPDATE_TRANSACTION'
+export const UPDATE_TRANSACTION_ERROR = PREFIX + 'UPDATE_TRANSACTION_ERROR'
 
-export const UPDATE_NATIVE_AMOUNT = PREFIX + 'UPDATE_NATIVE_AMOUNT'
+export const UPDATE_PARSED_URI_NATIVE_AMOUNT = PREFIX + 'UPDATE_NATIVE_AMOUNT'
+export const UPDATE_PARSED_URI_METADATA = PREFIX + 'UPDATE_METADATA'
 export const CHANGE_MINING_FEE = PREFIX + 'CHANGE_MINING_FEE'
 
-export type UpdateTransactionAction = {
-  type: typeof UPDATE_TRANSACTION,
-  data: {
-    transaction?: AbcTransaction | null,
-    error?: Error | null
-  }
+export type AbcMakeSpendInfo = {
+  currencyCode: string,
+  networkFeeOption?: string,
+  publicAddress?: string,
+  metadata?: any,
+  customNetworkFee?: any,
+  nativeAmount?: string,
+  spendTargets?: Array<AbcSpendTarget>
 }
 
 export const updateAmountSatoshi = (amountSatoshi: string) => ({
   type: UPDATE_AMOUNT_SATOSHI,
   data: {amountSatoshi}
 })
 
-export const updateTransactionAction = (
-  parsedUri: AbcParsedUri,
-  transaction: AbcTransaction | null,
-  error: Error | null): UpdateTransactionAction => ({
-    type: UPDATE_TRANSACTION,
-    data: {parsedUri, transaction, error}
-  })
+export const updateTransaction = (transaction: ?AbcTransaction): any => ({
```

Comment:
This is exactly what you do NOT want to do in redux. Do not break up one action into two if not necessary. This causes multiple renders.

Reply:
It's fixed in the new version I'm currently testing